src/compile/data/parse.ts

```diff
@@ -202,6 +204,10 @@ export function parseData(model: Model): DataComponent {
     }
 
     head = TimeUnitNode.makeFromEncoding(head, model) || head;
+
+    for (const calculate of CalculateNode.makeAllForSortIndex(head, model)) {
```

Comment:
Do we really still need for loop here?  Can't we just set head to the last item in the array?

Reply:
So it seems like the constructor and assemble code for `CalculateNode` doesn't handle the case where it has to generate multiple nodes from one instance. I could change it so that the constructor doesn't require for loop, but that would affect the code we already have for general `CalculateNode`

Reply:
I don't understand.    Why would the following code be different from the for loop you have here?

```
const sortIndexNodes = CalculateNode.makeAllForSortIndex(head, model);
head = sortIndexNodes[sortIndexNodes.length - 1];
```

Can you explain more if I overlook anything.

Reply:
I think what I'm trying to do might be different here. I saw that @domoritz made some changes to the way `bin` and `timeUnit` nodes are created and assembled. Previously we had to iterate over for loop to construct nodes returned from methods like `makeAll`, but the new changes made it so that `makeAll` returns only one node with all the components inside a dictionary with field name as a key(instead of multiple nodes with separate components). So the constructor of `bin` and `timeUnit` now takes the parent and the dictionary as params.

Anyway, I will follow your suggestion since I think it doesn't make sense for `CalculateNode` to have the same behavior as `bin` and `timeUnit` node as I explained above.

Reply:
> Previously we had to iterate over for loop to construct nodes returned from methods like makeAll, but the new changes made it so that makeAll returns only one node with all the components inside a dictionary with field name as a key(instead of multiple nodes with separate components). So the constructor of bin and timeUnit now takes the parent and the dictionary as params.

I think the old loop is updating head like you do down here in the past, but it's kinda annoying because you need to update head and keep passing current head to the next one.  

> returns only one node with all the components inside a dictionary with field name as a key

I think your case is more like geopoint/ geojson transform.  Which basically updating parent within the `parseAll` method.   So I think you could follow the pattern and do 

```
head = CalculateNode.parseAllSortIndex(head, model);
```

where your `parseAllSortIndex` returns just the final head (parent)

cc: @domoritz  Please correct me if I'm wrong.